synapse/events/third_party_rules.py

```diff
@@ -36,79 +40,119 @@ def __init__(self, hs):
 
         if module is not None:
             self.third_party_rules = module(
-                config=config, http_client=hs.get_simple_http_client()
+                config=config, module_api=ModuleApi(hs, hs.get_auth_handler()),
             )
 
-    @defer.inlineCallbacks
-    def check_event_allowed(self, event, context):
+    async def check_event_allowed(self, event: EventBase, context: EventContext):
```

Comment:
```suggestion
    async def check_event_allowed(self, event: EventBase, context: EventContext) -> bool:
```

and so on (if others exist)